utils/fix-rules.py

```diff
@@ -244,6 +275,92 @@ def rewrite_value_remove_prefix(line):
     return key + ": " + new_value
 
 
+class CCEFile:
+    def __init__(self, project_root=None):
+        if not project_root:
+            project_root = os.path.join(
+                os.path.dirname(os.path.abspath(__file__)), "..")
+        self.project_root = project_root
+
+    @property
+    def absolute_path(self):
+        raise NotImplementedError()
+
+    def line_to_cce(self, line):
+        return line
+
+    def line_isnt_cce(self, cce, line):
+        return line != cce
+
+    def read_cces(self):
+        with open(self.absolute_path, "r") as f:
+            cces = f.read().splitlines()
+        return cces
+
+    def remove_cce_from_file(self, cce):
+        file_lines = self.read_cces()
+        lines_except_cce = [
+            line for line in file_lines
+            if self.line_isnt_cce(cce, line)
+        ]
+        with open(self.absolute_path, "w") as f:
+            f.write("\n".join(lines_except_cce) + "\n")
+
+    def random_cce(self):
+        cces = self.read_cces()
+        random.shuffle(cces)
+        return cces[0].strip()
+
+
+class RedhatCCEFile(CCEFile):
+    @property
+    def absolute_path(self):
+        return os.path.join(self.project_root, "shared", "references", "cce-redhat-avail.txt")
+
+
+CCE_POOLS["redhat"] = RedhatCCEFile
+
+
+def add_to_the_section(file_contents, yaml_contents, section, new_keys):
+    to_insert = []
+
+    sec_ranges = find_section_lines(file_contents, section)
+    if len(sec_ranges) != 1:
+        raise RuntimeError("could not find one section: %s"
+                           % section)
+
+    begin, end = sec_ranges[0]
+
+    assert end > begin, "We need at least one identifier there already"
+    template_line = str(file_contents[end - 1])
+    leading_whitespace = re.match(r"^\s*", template_line).group()
+    for key, value in new_keys.items():
+        to_insert.append(leading_whitespace + key + ": " + value)
+
+    new_contents = file_contents[:end] + to_insert + file_contents[end:]
+    return new_contents
+
+
+def sort_section(file_contents, yaml_contents, section):
```

Comment:
In #6882, we have:

https://github.com/ComplianceAsCode/content/pull/6882/commits/8e115a101af074a7d03de3f8bbf1fed34c29fb7d#diff-7e689958b33c2feea57ead09db20cb40ea8dc8ee5a20afdffec67ede2c8b4c25R226

`ssg.rule_yaml.sort_section_keys(...)`

You might consider seeing if we could have a single implementation for use by both scripts?